### PR TITLE
Functional Tests - Product settings choose quantity discounts based on

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Product settings choose quantity discounts based on
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH=TEST_PATH="functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17699)

### CODE REVIEW

File: tests/puppeteer/pages/FO/FObasePage.js [hunk 1]
```diff
@@ -149,4 +150,15 @@ module.exports = class Home extends CommonPage {
...
+
+  /**
+   * Go to cart page
+   * @returns {Promise<void>}
+   */
+  async goToCartPage() {
+    await Promise.all([
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
+      this.page.click(this.cartLink),
+    ]);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/FObasePage.js]: please use `clickAndWaitForNavigation` function

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 2]
```diff
@@ -62,15 +68,15 @@ module.exports = class AddProduct extends BOBasePage {
    * @param switchProductOnline
    * @return {Promise<textContent>}
    */
-  async createEditProduct(productData, switchProductOnline = true) {
+  async createEditProduct(productData, switchProductOnline = true, withSpecificPrice = false) {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/catalog/products/add.js]: please add description for the new param

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  // Login into BO and go to product settings page
+  loginCommon.loginBO();
+
+  it('should go to \'Shop parameters > Product Settings\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductSettingsPage1', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js]: @boubkerbribri To remove ?

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  it('should choose quantity discounts based on \'Combinations\'', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'chooseQuantityDiscountsBasedOnCombinations', baseContext);
+    const result = await this.pageObjects.productSettingsPage.chooseQuantityDiscountsBasedOn('Combinations');
+    await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+  });
+
+  it('should view my shop, add first and second combination to the cart and check price TTC in FO', async function () {
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js]:
```suggestion
  it('should view my shop and check price TTC in FO', async function () {
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js]: no, it's the first one in the test, we need it, maybe after adding [this](https://github.com/PrestaShop/PrestaShop/issues/17517), we can delete all of them from tests

Comment (Progi1984) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/04_quantityDiscountsBasedOn.js]: Thanks ;)

**Review by boubkerbribri — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**